src/ui/components/ShowMoreCard/index.js

```diff
@@ -1,55 +1,66 @@
+/* @flow */
 /* eslint-disable react/no-danger */
-
+import invariant from 'invariant';
 import makeClassName from 'classnames';
 import * as React from 'react';
-import PropTypes from 'prop-types';
 import { compose } from 'redux';
 
 import translate from 'core/i18n/translate';
+import withUIState from 'core/withUIState';
 import { sanitizeHTML } from 'core/utils';
 import Card from 'ui/components/Card';
+import type { I18nType } from 'core/types/i18n';
 
 import './styles.scss';
 
 // This refers to height of `Card-contents` CSS class,
 // beyond which it will add read more link.
 export const MAX_HEIGHT = 150;
 
-export class ShowMoreCardBase extends React.Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-    header: PropTypes.node,
-    i18n: PropTypes.object.isRequired,
-  };
+type UIStateType = {|
+  expanded: boolean,
+|};
 
-  constructor(props) {
-    super(props);
-    this.state = { expanded: true };
-  }
+type Props = {|
+  children: React.Element<any> | string,
+  className?: string,
+  header?: React.Element<any> | string,
+  id: string,
+|};
+
+type InternalProps = {|
+  ...Props,
+  i18n: I18nType,
+  setUIState: ($Shape<UIStateType>) => void,
+  uiState: UIStateType,
+|};
 
-  componentWillReceiveProps() {
```

Comment:
I decided to remove this b/c with these updates, it now triggers every time setUiState is called ....so it causes an endless loop. It seemed like in one of the tests below that there is a possibility that the child could change and we want to see the truncatetoMaxHeight get called for this but I still was questioning whether this was something we needed..? Next time the page loads, it should get updated (I think).

let me know if you see any issues with this..

Reply:
hmm..I guess I could check if there are prop changes to anything besides the uiState...

Reply:
for now tho, I am thinking I will as I have it. I am still not convinced the props will be changing too often...

Reply:
I've looked into this so more and I was not able to get this to work :/ 
When I leave this in componentDidMount the height on pages that are not the original page is only about 19px :/  Even tho when I inspect the node it I see the correct height there..so I am not entirely sure why it's showing the 19px. I see it get updated to the accurate amount on one of the other/later component life cycles (such as componentWillReceiveProps or shouldComponentUpdate etc, but if you call truncate from here, it updates the uiState prop and runs through this again (loop).